# Regression HW 3

SEOYEON CHOI  
2022-11-21

고급회귀분석 과제, CH03,07

------------------------------------------------------------------------

고급회귀분석 세번째 과제입니다.

제출 기한 : 11월 21일

제출 방법

-   직접 제출(607호) 도 가능하지만,

-   문서 작성 후 pdf로 변환(★★★)하여 lms에 제출을 추천

(pdf 아닌 문서는 미제출로 간주)

주의사항

-   pdf로 꼭 변환하여 제출

-   풀이가 꼭 있어야 함 (답만 적혀 있는 경우 ’0’점 처리)

-   부정행위 시 ’F’학점

-   계산은 R로 해도 되지만 계산 풀이 과정이 꼭 있어야 함!!

예) R에서 lm으로 beta의 추정량을 구하면 안 됨. 수업 시간에 배운 식으로
풀이를 적어야 함.

-   행렬식 계산은 R로 해도 됩니다.

\*\*\*\*\*\*\*\*\*\*\*\* R을 이용해서 푸는 문제는, R 코드도 같이 업로드.

------------------------------------------------------------------------

# 1.

어떤 큰 공장에서 동일한 기계들의 정비기록에 관한 표본자료를 취하였다.
이는 기계의 사용연도(age of machines)와 정비비용(maintenance cost) 간에
어떤 관계가 있는가를 밝혀내기 위한 것이다. 그 자료는 다음과 같다 (표본의
크기 $n = 14$).

| 사용연도$X$(단위:년) | 정비비용$Y$(단위:1,000원) |
|:--------------------:|:-------------------------:|
|          3           |            39             |
|          1           |            24             |
|          5           |            115            |
|          8           |            105            |
|          1           |            50             |
|          4           |            86             |
|          2           |            67             |
|          6           |            90             |
|          9           |            140            |
|          3           |            112            |
|          5           |            70             |
|          7           |            186            |
|          2           |            43             |
|          6           |            126            |

In [1]:
dt <- data.frame(y=c(39,24,115,105,50,86,67,90,140,112,70,186,43,126), 
                 x=c(3,1,5,8,1,4,2,6,9,3,5,7,2,6))

In [2]:
x <- matrix(c(rep(1,14), dt$x), ncol=2)

In [3]:
y <- matrix(c(dt$y), ncol=1)

## (1)

$X^\top X, X^\top y, y^\top y$와 $(X^\top X)^{-1}$을 구하시오.

In [4]:
xx <- t(x) %*% x

In [5]:
xx

In [6]:
xy <- t(x) %*% y

In [7]:
xy

In [8]:
yy <- t(y) %*% y

In [9]:
yy

In [10]:
xx_i <- solve(xx)

In [11]:
xx_i

## (2)

$\hat{\beta} = (X^\top X)^{-1}X^\top y$를 구하고, 적합된 회귀선형을 써
보아라.

In [12]:
betahat <- xx_i %*%  xy

In [13]:
betahat

$y = 29.1 + 13.6x$

In [14]:
coef(lm(y~x, dt))

(Intercept) 29.1070234113712 x 13.6371237458194

## (3)

$\sigma^2$을 MSE 로 추정할 경우, $\hat{\beta}$의 분산-공분산행렬의
추정은 $\hat{Var}(\hat{\beta}) = (X^\top X)^{-1}$(MSE)이다. 먼저
분산분석하여 MSE 를 구하고 $\hat{Var}(\hat{\beta})$을 구하시오

In [15]:
n <- 14

In [16]:
p <- 1

In [17]:
In = diag(rep(1,n))

In [18]:
H = x %*% xx_i %*% t(x)

$SSE = y^\top(I-H)y$

$SST = y^\top y - n(\bar{y})^2$

In [19]:
sse = t(y) %*% (In-H) %*% y
sse

In [20]:
sst = t(y) %*% y - n * mean(y)^2
sst

In [21]:
ssr = sst- sse
ssr

In [22]:
msr = ssr/p
msr

In [23]:
mse = sse/(n-p-1)
mse

In [24]:
f0 = msr/mse
f0

In [25]:
qf(0.95,p,n-p-1)

4.74722534672251

**분산분석표**

| 요인 |  제곱합  | 자유도(df) | 평균제곱(MS) |  $F_0$   | 유의확률 |
|:----:|:--------:|:----------:|:------------:|:--------:|:--------:|
| 회귀 | 15887.25 |     1      |   15887.25   | 18.75293 | 4.747225 |
| 잔차 | 10166.25 |     12     |    847.19    |          |          |
|  계  | 26053.5  |     13     |              |          |          |

In [26]:
hat_var_betahat <- mse[,] * xx_i

In [27]:
vcov(lm(y~x, dt))

# 2.

어떤 공정에서 나오는 제품의 강도$(kg/cm^2)$가 그 공정의 온도와 압력에
어떠한 영향을 받는가를 조사하기 위하여 다음의 데이터를 얻었다.

| 공정온도$x_1$(단위:°C) | 공정압력$x_2$(단위:$psi$) | 강도$y$(단위:$kg/cm^2$) |
|:----------------------:|:----------------------:|:----------------------:|
|          195           |            57             |          81.4           |
|          179           |            61             |          122.2          |
|          205           |            60             |          101.7          |
|          204           |            62             |          175.6          |
|          201           |            61             |          150.3          |
|          184           |            54             |          64.8           |
|          210           |            58             |          92.1           |
|          209           |            61             |          113.8          |

In [28]:
dt <- data.frame(y=c(81.4,122.2,101.7,175.6,150.3,64.8,92.1,113.8),
                 x1 = c(195,179,205,204,201,184,210,209),
                 x2 = c(57,61,60,62,61,54,58,61))

In [29]:
m2 <- lm(y~., dt)

In [30]:
summary(m2)


Call:
lm(formula = y ~ ., data = dt)

Residuals:
      1       2       3       4       5       6       7       8 
 -5.259 -14.775 -18.742  31.259  17.279  11.744  -3.723 -17.783 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)  
(Intercept) -554.3112   196.9775  -2.814   0.0374 *
x1            -0.1797     0.7626  -0.236   0.8230  
x2            11.8600     3.2301   3.672   0.0144 *
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 21.64 on 5 degrees of freedom
Multiple R-squared:  0.7478,    Adjusted R-squared:  0.6469 
F-statistic: 7.412 on 2 and 5 DF,  p-value: 0.03195

In [31]:
anova(m2)

## (1)

선형회귀모형,
$y_j = \beta_0 + \beta_1 x_{1j} + \beta_2 x_{2j} + \epsilon_j$ 가
성립된다고 가정하고 데이터로부터 회귀모형을 추정하시오.

In [32]:
x = matrix(c(rep(1,8),dt$x1, dt$x2), ncol=3)

In [33]:
y = dt$y

In [34]:
xx <- t(x) %*% x
xx

In [35]:
xx_i <- solve(xx)
xx_i

In [36]:
xy <- t(x) %*% y
xy

In [37]:
yy <- t(y) %*% y
yy

In [38]:
b <- xx_i %*% xy
b

$y = -554.31 - 0.1797 x_1 + 11.86 x_2$

## (2)

오차분산이 $\sigma^2 = 3$이라 하면,
$Var(\hat{\beta}_0), Var(\hat{\beta}_1), Var(\hat{\beta}_2)$와
$Cov(\hat{\beta}_1, \hat{\beta}_2)$는 무엇인가?

$var(\hat{\beta}) = (x^\top x)^{-1} \sigma^2$

In [39]:
xx_i * 3

$var(\hat{\beta}_0) = 248.6250$

$var(\hat{\beta}_1) = 0.0037$

$var(\hat{\beta}_2) = 0.0669$

$cov(\hat{\beta}_1, \hat{\beta}_2) = -0.0057$

## (3)

$x_1 = 200$°C이고 $x_2 = 59\text{psi}$에서 평균 제품의 강도의 추정값
$\hat{y}$는 얼마인가? 이의 분산을 $\sigma^2 = 3$이라 가정하고 구하시오.

In [40]:
new_dt <- data.frame(x1=200, x2=59)
new_dt

In [41]:
predict(m2, newdata = new_dt)

1: 109.480424963516

In [42]:
x0 <- c(1,200,59)
x0

1 200 59

In [43]:
mu0 <- x0 %*% b
mu0

$\hat{\mu}_0 = 109.4804$

## (4)

추정된 회귀계수 $\hat{\beta}_1, \hat{\beta}_2$의 의미는 무엇인가?

$\hat{\beta}_1$ : 공정압력이 일정하게 유지되었을 때, 공정온도가 1도씨
증가하면 강도는 0.1797만큼 감소한다.

$\hat{\beta}_2$ : 공정온도가 일정하게 유지되었을 때, 공정압력이 1psi
증가하면 강도는 11.8600만큼 증가한다.

## (5)

분산분석표를 작성하고 $\alpha = 0.05$로 $F$-검정을 행하시오.

In [44]:
anova(m2)

In [45]:
I8 = diag(rep(1,8))
I8

In [46]:
H = x %*% xx_i %*% t(x)
H

$SSE = y^\top (I-H)y$

$SST = y^\top y - n(\bar{y})^2$

In [47]:
sse = t(y) %*% (I8-H) %*% y
sse

In [48]:
sst = t(y) %*% y - 8 * mean(y)^2
sst

In [49]:
ssr = sst- sse
ssr

In [50]:
msr = ssr/2
msr

In [51]:
mse = sse/5
mse

In [52]:
f0 = msr/mse
f0

In [53]:
qf(0.95,2,5)

5.78613504334997

| 요인 |  제곱합  | 자유도(df) | 평균제곱(MS) | $F_0$  | 유의확률 |
|:----:|:--------:|:----------:|:------------:|:------:|:--------:|
| 회귀 | 6940.195 |     2      |   3470.097   | 7.4119 | 5.786135 |
| 잔차 | 2340.884 |     5      |   468.177    |        |          |
|  계  | 9281.079 |     7      |              |        |          |

결론 : 검정통계량의 관측값이 기각역에 속하므로 귀무가설 기각. 즉 모형이
유의수준 0.05하에서 유의하다고 할 수 있다.

## (6)

결정계수 $R^2$을 구하시오.

In [54]:
summary(m2)$r.squared

0.747778895028607

$R^2 = \frac{SSR}{SST}$

In [55]:
ssr/sst
# 0.7478

## (7)

수정된 결정계수 $R^2_{adj}$ 을 구하시오.

In [56]:
summary(m2)$adj

0.64689045304005

$R^2_{adj} = 1-\frac{SSE(n-p-1)}{SST/(n-1)}$

In [57]:
1-(sse/5)/(sst/7)
# 0.6469

## (8)

$\sigma^2$의 추정값 MSE 를 구하시오.

$\sigma^2 = MSE$

In [58]:
mse
##468.17687

## (9)

MSR의 기대값을 $\sigma^2$과 $\beta_0, \beta_1, \beta_2$의 함수로
표시하여라.

In [59]:
I8 <- diag(rep(1,8))
I8

In [60]:
one1 <- rep(1,8)
one1

1 1 1 1 1 1 1 1

In [61]:
Jn <- one1 %*% t(one1)
Jn

In [62]:
t(x) %*% (I8 - Jn/8) %*% x

# 3.

어떤 공장에서 물의 소비량을 조사하기 위하여 매달의 물소비량($y$),
평균온도($x_1$), 작업일수($x_2$)와 작업량($x_3$)에 관한 데이터를 얻었다.

| 물소비량$y$(단위:1,000톤) | 평균온도$x_1$(단위:°C) | 작업일수$x_2$(단위:일) | 작업량$x_3$(단위:1,000톤) |
|:----------------:|:----------------:|:----------------:|:----------------:|
|            2.8            |           10           |           27           |            64             |
|            3.9            |           24           |           26           |            72             |
|            3.9            |           25           |           28           |            80             |
|            4.4            |           28           |           26           |            88             |
|            3.1            |           15           |           30           |            81             |
|            3.1            |           18           |           24           |            45             |
|            3.5            |           22           |           27           |            46             |
|            3.6            |           22           |           25           |            69             |
|            3.0            |           12           |           27           |            54             |
|            3.3            |           15           |           25           |            39             |

In [63]:
dt <- data.frame(y=c(2.8,3.9,3.9,4.4,3.1,3.1,3.5,3.6,3.0,3.3),
                 x1 = c(10,24,25,28,15,18,22,22,12,15),
                 x2 = c(27,26,28,26,30,24,27,25,27,25),
                 x3 = c(64,72,80,88,81,45,46,69,54,39))

In [64]:
m3 <- lm(y~., dt)

In [65]:
summary(m3)


Call:
lm(formula = y ~ ., data = dt)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.23490 -0.07744 -0.02166  0.08840  0.23442 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)   
(Intercept)  2.409213   1.125954   2.140  0.07618 . 
x1           0.069788   0.012640   5.521  0.00149 **
x2          -0.024767   0.044830  -0.552  0.60060   
x3           0.005864   0.005052   1.161  0.28978   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.172 on 6 degrees of freedom
Multiple R-squared:  0.9202,    Adjusted R-squared:  0.8803 
F-statistic: 23.05 on 3 and 6 DF,  p-value: 0.001079

In [66]:
anova(m3)

## (1)

데이터로부터 회귀모형,
$\hat{y} = \hat{\beta}_0 + \hat{\beta}_1 x_1 + \hat{\beta}_2 x_2 + \hat{\beta}_3 x_3$
을 구하여라. 그리고 어째서 이 모형이 선택되었는가에 대하여 토의하시오.

In [67]:
x = matrix(c(rep(1,10),dt$x1, dt$x2, dt$x3), ncol=4)

In [68]:
y = dt$y

In [69]:
xx <- t(x) %*% x
xx

In [70]:
xx_i <- solve(xx)
xx_i

In [71]:
xy <- t(x) %*% y
xy

In [72]:
yy <- t(y) %*% y
yy

In [73]:
b <- xx_i %*% xy
b

$\hat{y} = 2.409 + 0.0698x_1 - 0.0248x_2 + 0.0059x_3$

## (2)

$\hat{\beta}_1, \hat{\beta}_2$와 $\hat{\beta}_3$의 의미는 무엇인가?

$\hat{\beta}_1$ : 작업일수, 작업량이 변하지 않을 때, 평균온도가 1도씨
증가하면 물소비량은 69.8톤 증가한다.

$\hat{\beta}_2$ : 평균온도, 작업량이 변하지 않을 때, 작업일수가 1일
증가하면 물소비량은 24.8톤 감소한다.

$\hat{\beta}_3$ : 평균온도, 작업일수이 변하지 않을 때, 작업량이 1000톤
증가하면 물소비량은 5.9톤 증가한다.

## (3)

$Var(\hat{\beta}_3)$을 구하시오. $\sigma^2$을 MSE 로 추정하면
$\hat{Var}(\hat{\beta}_3)$은 무엇인가?

In [74]:
summary(m3)$coefficients

$var(\beta_3) = (x^\top x)^{-1}_{(4,4)} \sigma^2$

In [75]:
xx_i[4,4]
## 0.0009*sigma^2

0.000862438702127396

In [76]:
hat_var_b3 <- xx_i[4,4] * mse
hat_var_b3

$var(\hat{\beta}_3) = 0.0000255$

## (4)

분산분석표를 작성하고, 결정계수 $R^2$을 구하시오.

In [77]:
anova(m3)

In [78]:
I10 = diag(rep(1,10))
I10

In [79]:
H = x %*% xx_i %*% t(x)
H

$SSE = y^\top (I-H)y$

$SST = y^\top y - n(\bar{y})^2$

In [80]:
sse = t(y) %*% (I10-H) %*% y
sse

In [81]:
sst = t(y) %*% y - 10 * mean(y)^2
sst

In [82]:
ssr = sst- sse
ssr

In [83]:
msr = ssr/3
msr

In [84]:
mse = sse/6
mse

In [85]:
f0 = msr/mse
f0

In [86]:
qf(0.95,3,6)

4.75706266308941

| 요인 | 제곱합 | 자유도(df) | 평균제곱(MS) |  $F_0$  | 유의확률 |
|:----:|:------:|:----------:|:------------:|:-------:|:--------:|
| 회귀 | 2.0465 |     3      |    0.6822    | 23.0544 |  4.7571  |
| 잔차 | 0.1775 |     6      |    0.0296    |         |          |
|  계  | 2.2240 |     9      |              |         |          |

$R^2$

In [87]:
ssr/sst
#0.9202

## (5)

$x_1 = 20, x_2 = 27, x_3 = 60$에서 평균 물소비량을 추정하시오. 이 추정된
소비량의 표준편차의 추정값을 구하시오.

In [88]:
new_dt <- data.frame(x1=20, x2=27, x3 = 60)
new_dt

In [89]:
predict(m3, newdata = new_dt)

1: 3.48814105556645

In [90]:
x0 <- c(1,20,27,60)
x0

1 20 27 60

In [91]:
mu0 <- x0 %*% b
mu0

$\hat{\mu}_0 = 3.4881$

In [92]:
var_mu0 <- (t(x0) %*% xx_i %*% x0) * mse
var_mu0

In [93]:
sqrt(var_mu0)
# se(hat mu0) = 0.0712

## (6)

$x_1 = 20, x_2 = 27, x_3 = 60$에서 어느 한 달의 물소비량 $y_s$ 를
예측하여라. 이 예측된 물소비량의 표준편차의 추정값을 구하시오.

In [94]:
x0 <- c(1,20,27,60)
x0

1 20 27 60

In [95]:
mu0 <- x0 %*% b
mu0

$\hat{y}_s = 3.4881$

In [96]:
var_ys <- (1+t(x0) %*% xx_i %*% x0) * mse
var_ys

In [97]:
sqrt(var_ys)
# se(hat ys) = 0.1862

## (7)

$y_j = \beta_0 + \beta_1 x_{1j} + \beta_2 x_{2j} + \epsilon_j$ 라
가정하고 회귀제곱합 SSR의 기대값을 $\sigma^2$과
$\beta_i, i = 0, 1, 2, 3$의 함수로 표시하여라.

In [98]:
I10 <- diag(rep(1,10))
I10

In [99]:
one1 <- rep(1,10)
one1

1 1 1 1 1 1 1 1 1 1

In [100]:
Jn <- one1 %*% t(one1)
Jn

In [101]:
t(x) %*% (I10 - Jn/10) %*% x

## (8)

$\beta_3$의 95% 신뢰구간을 구하시오. 이 신뢰구간의 의미를 해석하시오.

In [102]:
confint(m3)

$\beta_3 \pm t_{0.025}(6) se(\hat{\beta}_3)$

In [103]:
b[4] - qt(0.975,6) * sqrt(hat_var_b3)

In [104]:
b[4] + qt(0.975,6) * sqrt(hat_var_b3)

In [105]:
#(-0.0065, 0.0182)

이 신뢰구간이 $\beta_3$을 포함하고 있을 것이라고 95% 확신한다.

## (9)

$\beta_1$의 99% 신뢰구간을 구하시오. 이 신뢰구간의 의미를 해석하시오.

In [106]:
confint(m3, level=0.99)

$\beta_1 \pm t_{0.005}(6) se(\hat{\beta}_1)$

In [107]:
hat_var_b1 = xx_i[2,2] * mse
hat_var_b1

In [108]:
b[2] - qt(0.995,6) * sqrt(hat_var_b1)

In [109]:
b[2] + qt(0.995,6) * sqrt(hat_var_b1)

In [110]:
#(0.0229, 0.1167)

이 신뢰구간이 $\beta_1$을 포함하고 있을 것이라고 99% 확신한다.

## (10)

$x_1 = 20, x_2 = 27, x_3 = 60$에서 평균 물소비량, $E(y)$의 95%
신뢰구간을 구하시오.

In [111]:
predict(m3, newdata = new_dt, interval = c("confidence"))

In [112]:
mu0 - qt(0.975,6) * sqrt(var_mu0)

In [113]:
mu0 + qt(0.975,6) * sqrt(var_mu0)

In [114]:
#(3.3140, 3.6623)

## (11)

가설 $H_0 : \beta_1 = 0, H_1 : \beta_1 > 0$을 $\alpha = 0.05$로
검정하시오.

In [115]:
summary(m3)


Call:
lm(formula = y ~ ., data = dt)

Residuals:
     Min       1Q   Median       3Q      Max 
-0.23490 -0.07744 -0.02166  0.08840  0.23442 

Coefficients:
             Estimate Std. Error t value Pr(>|t|)   
(Intercept)  2.409213   1.125954   2.140  0.07618 . 
x1           0.069788   0.012640   5.521  0.00149 **
x2          -0.024767   0.044830  -0.552  0.60060   
x3           0.005864   0.005052   1.161  0.28978   
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 0.172 on 6 degrees of freedom
Multiple R-squared:  0.9202,    Adjusted R-squared:  0.8803 
F-statistic: 23.05 on 3 and 6 DF,  p-value: 0.001079

검정통계량 : $t_0 = \frac{\hat{\beta}_1}{se(\hat{\beta}_1)}$

In [116]:
t0 <- b[2]/sqrt(hat_var_b1)
t0

$t_0 = 5.5211$

기각역 : $t_{0.05}(6) = 1.9432$

In [117]:
qt(0.95,6)

1.9431802805153

결론 : $H_0$ 기각 가능. 즉 유의수준 5%에서 $\beta_2$는 0보다 크다고 할
수 있다.

$T = \frac{\hat{\beta}_i - \beta^0_i}{\hat{\sigma} \sqrt{c_{ii}}}$

## (12)

가설 $H_0 : \beta_1 = \beta_2 = \beta_3$을 $\alpha = 0.05$로 검정하시오.

In [118]:
library(car) 

Loading required package: carData


In [119]:
linearHypothesis(m3, matrix(c(0,1,-1,0,0,0,1,-1), byrow = T, nrow=2),c(0,0))

$H_0 : \beta_1 = \beta_2 = \beta_3$

$T= (0,1,-1,0) (0,0,1,-1)$

$y= \beta_0 + \beta_1 x_1 + \beta_1 x_2 + \beta_1 x_3 + \epsilon = \beta_0 + \beta_1(x_1+x_2+x_3)$

In [120]:
z <- matrix(c(rep(1,10), dt$x1+dt$x2+dt$x3), ncol=2)
z

In [121]:
zz <- t(z) %*% z
zz

In [122]:
zz_i <- solve(zz)
zz_i

In [123]:
zy <- t(z) %*% y
zy

In [124]:
yy <- t(y) %*% y
yy

In [125]:
z %*% zz_i %*% t(z)

In [126]:
gammahat <- zz_i %*% t(z) %*% y
gammahat

In [127]:
e <- y - z %*% gammahat
e

In [128]:
sse_rm <- t(e) %*% e
sse_rm

In [129]:
f0 = ((sse_rm - sse)/(2))/(sse/6)
f0

In [130]:
qf(0.95,2,6)

5.14325284978472

검정통계량 : $\frac{(SSE_{RM} - SSE_{FM})/r}{SSE_{FM}/(n-p-1)}$

관측값 $f_0 = 15.968$

기각역 : $f_0 > 5.1433$

결론 : 기각할 수 있다.

## (13)

가설 $H_0 : \beta_1 = \beta_2 + 3$을 $\alpha = 0.05$로 검정하시오.

$H_0 : \beta_1 = \beta_2 = \beta_3$

$T= (0,1,-1,0)$

$y = \beta_0 + (\beta_2+3)x_1 + \beta_2 x_2 + \beta_3 x_3 + \epsilon$
$y - 3x_1 = \beta_0 + \beta_2(x_1 + x_2) + \beta_3 x_3 + \epsilon$

In [131]:
yy <- y-3*dt$x1
yy

-27.2 -68.1 -71.1 -79.6 -41.9 -50.9 -62.5 -62.4 -33 -41.7

In [132]:
z <- matrix(c(rep(1,10), dt$x1+dt$x2,dt$x3), ncol=3)
z

In [133]:
zz <- t(z) %*% z
zz

In [134]:
zz_i <- solve(zz)
zz_i

In [135]:
gammahat <- zz_i %*% t(z) %*% yy
gammahat

In [136]:
e <- yy - z %*% gammahat
e

In [137]:
sse_rm <- t(e) %*% e
sse_rm

In [138]:
f0 = ((sse_rm - sse)/(1))/(sse/6)
f0

In [139]:
qf(0.95,1,6)

5.9873776072737

검정통계량 : $\frac{(SSE_RM - SSE_FM)/r)}{(SSE_FM/(n-p-1))}$

관측값 $f_0 = 5311.082$

기각역 : $f_0 > 5.9874$

결론 : 기각할 수 있다.

## (14)

$x_1 = 20, x_2 = 27, x_3 = 60$에서 $E(y)$에 관한 가설
$H_0 : E(y) = 3.5, H_1 : E(y) ̸\neq 3.5$를 $\alpha = 0.05$로 검정하시오.

sy 풀이

$T_0 = \frac{\hat{y} - E(y)}{\sqrt{Var{\hat{y}}}} = \frac{\hat{y} - E(Y)}{\sqrt{x^\top (X^\top X)^{-1} x MSE}} \sim t(n-1)$

In [140]:
x_indi <- matrix(c(1,20,27,60),nrow=1,ncol=4)

In [141]:
yhat <- x_indi %*% b
round(yhat,5)

$Var{y} = x_i (X^\top X)^{-1} x_i^\top \sigma^2$

In [142]:
var_y <- ((x_indi %*%solve(t(x) %*% x) %*% t(x_indi) ) * mse)[,]
round(var_y,5)

0.00507

In [143]:
t_0 <- (yhat - 3.5)/sqrt(var_y)
round(t_0[,],5)

-0.16663

In [144]:
round(-pt(0.025,9),5)

-0.5097

$F_0=-0.16663$ \> $t_{0.025}(9)=-0.5097$ 로 귀무가설을 기각하지 못하여
$E(y) = 3.5$라는 결론이 나온다.

# 4.

중회귀모형, $y = X\beta + \epsilon, \epsilon \sim N(0_n, I\sigma^2)$에서
$X$ 가 $n \times (p + 1)$행렬이고 rank가 $p + 1$이라면 적합된 모형
$\hat{y} = X\hat{\beta}$에 대하여
$$\sum^{n}_{j=1} Var(\hat{y}_j ) = (p + 1)\sigma^2$$ 이 됨을 증명하여라.

$\hat{\bf{y}}= X\hat{\beta} = X(X^\top X)^{-1} X^\top y = Hy$

$Var(X\hat{\beta}) = XVar(\hat{\beta})X^\top$

$Var(\hat{\beta}) = Var((X^\top X)^{-1} X^\top y) = (X^\top X)^{-1} X^\top Var(y)X(X^\top X)^{-1} = (X^\top X)^{-1} X^\top X(X^\top X)^{-1} \sigma^2 = (X^\top X )^{-1} \sigma^2$

$Var(\hat{y}) = Var(X\hat{\beta}) = X(X^\top X)^{-1} X^\top \sigma^2$

$\to Var(\hat{y}) = HVar(y) H^\top = HH\sigma^2 = H\sigma^2$

$\star H^\top = H, H^2 = H, Var(y) = I_n \sigma^2$

$\sum^{n}_{j=1} Var(\hat{y}_j) = tr(Var(\hat{y})) = tr(H\sigma^2) = \sigma^2 tr(H) = tr((X^\top X)^{-1} X^\top X)\sigma^2 = tr(I_{p+1})\sigma^2 = (p+1)\sigma^2$

$\star tr(X (X^\top X)^{-1} X^\top)$

# 5.

중회귀모형에서 다음을 증명하시오.

## (1)

$Cov(e, y) = \sigma^2[I_n − X(X^\top X)^{-1}X^\top]$

$\bf{e} = y - \hat{y} = y - Hy = (I-H)y$

$Cov(\bf{e},y) = Cov((I-H)y,y) = (I-H)Var(y) = (I-H) \sigma^2$

$\star Cov(Ax,y) = A Cov(X,Y)$

$\star Var(y) = \sigma^2$

## (2)

$Cov(e, \hat{y}) = \mathbb{O}_n$

$Cov(e(I - H)\bf{y},Hy) = (I-H) Cov(y,y)H^\top = (I-H)H\sigma^2 = \mathbb{O}_n \sigma^2$

$\star Cov(Ax,By) = ACov(X,Y)B^\top$

$\star H^\top= H$

$\star H^2 = H$

## (3)

$Cov(e, \hat{\beta}) = \mathbb{O}_{n\times(k+1)}$

$Cov((I-H)\bf{y},(X^\top X)^{-1} y) = (I-H) Cov(y,y) X (X^\top X)^{-1} = (I - X(X^\top X)^{-1} X^\top ) X (X^\top X)^{-1} \sigma^2 = \{ X(X^\top X)^{-1} - X(X^\top X)^{-1} \} \sigma^2 = \mathbb{O}_{n \times ([+1)}$

## (4)

$Cov(\epsilon, \hat{\beta}) = \sigma^2 X(X^\top X)^{-1}$

$\bf{\hat{\beta}} = (X^\top X)^{-1} X^\top y = (X^\top X)^{-1} X^\top (X\beta + \epsilon) = \beta + (X^\top X)^{-1} X^\top \epsilon$

$Cov(\bf{\epsilon} , \beta + (X^\top X)^{-1} X^\top \epsilon ) = Cov(\epsilon, \beta) + Cov(\epsilon, \epsilon)X(X^\top X)^{-1} = \sigma^2 X(X^\top X)^{-1}$

$\star Cov(\epsilon, \beta) = 0$

$\star Cov(\epsilon, \epsilon) = I_n \sigma^2$

## (5)

$\sum^{n}_{j=1} e_j y_j = SSE$

$\sum^{n}_{j=1} e_j y_j = \bf{e^\top y} = \{ (I - H)y\}^\top y = y^\top (I-H)y$

$\star \bf{e}^\top (e_1, \dots , e_n) = (y-\hat{y})^\top$

$\star \bf{y}^\top = (y_1 , \dots , y_n)$

$SSE = \sum^n_{j=1} (y_i - \hat{y}_j)^2 = (\bf{y} - \hat{y} ) ^\top ( y - \hat{y} ) = e^\top e = \{ (I - H)y \}^\top \{ (I - H)y \} = y^\top (I - H) (I-H)y = y^\top (I - H)y$

$\star I - H_H+H^2 = I-H, H^2 = H$

$\therefore \sum^{n}_{j=1} e_j y_j = SSE$

## (6)

$\sum^{n}_{j=1} e_j \hat{y}_j = 0$

$\sum^{n}_{j=1} \bf{e_j \hat{y}_j} = e^\top \hat{y} = y^\top (I-H) Hy = y^\top_{1\times n} \mathbb{O}_{n\times n} y_{n \times 1} = 0$

$\star H - H^2 = H - H = \mathbb{O}_{n \times n}$

# 1 plus.

다음의 데이터에 대하여 모형
$y_i = \beta_0 + \beta_1 x_{1j} + \beta_2 x_{2j} + \epsilon_j, \epsilon \sim N(0,\sigma^2)$이
옳다고 가정하고 질문에 답하여라.

|  y  |  1  |  5  |  0  |  4  |  4  | -1  |
|:---:|:---:|:---:|:---:|:---:|:---:|:---:|
| x_1 |  1  |  2  |  1  |  3  |  3  |  3  |
| x_2 |  1  |  1  |  2  |  1  |  2  |  3  |

In [145]:
library(car)

In [146]:
y <- c(1,5,0,4,4,-1)

In [147]:
x <- matrix(c(1,1,1,1,1,1,1,2,1,3,3,3,1,1,2,1,2,3), ncol=3)

In [148]:
dt <- data.frame(y=y,x1=x[,2],x2 = x[,3])

In [149]:
m <- lm(y~., dt)

In [150]:
summary(m)


Call:
lm(formula = y ~ ., data = dt)

Residuals:
      1       2       3       4       5       6 
-1.1395  1.3488  0.4651 -1.1628  1.4419 -0.9535 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   3.2326     1.9740   1.638   0.2000  
x1            1.5116     0.7713   1.960   0.1449  
x2           -2.6047     0.9288  -2.804   0.0676 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.599 on 3 degrees of freedom
Multiple R-squared:  0.7511,    Adjusted R-squared:  0.5852 
F-statistic: 4.527 on 2 and 3 DF,  p-value: 0.1242

In [151]:
anova(m)

## (1)

$X^\top X, X^\top y y^\top y$와 $(X^\top X)^{-1}$ 을 구하시오.

In [152]:
xx <- t(x) %*% x
xx

In [153]:
xx_i <- solve(xx)
xx_i

In [154]:
xx_i2 <- round(xx_i,2)
xx_i2

In [155]:
xy <- t(x) %*% y
xy

In [156]:
yy <- t(y) %*% y
yy

## (2)

$\beta_0, \beta_1, \beta_2, \sigma^2$를 추정하시오.

In [157]:
coef(m)

(Intercept) 3.23255813953489 x1 1.51162790697674 x2 -2.6046511627907

In [158]:
anova(m)$Mean[3]

2.55813953488372

In [159]:
b <- xx_i %*% xy
b

In [160]:
b1 <- round(round(xx_i,2)%*% xy,2)
b1

In [161]:
haty <- x %*% b
haty

In [162]:
haty_2 <- round(x %*% b1,2)
haty_2

In [163]:
e <- y - haty
e

In [164]:
e_2 <- y - haty_2
e_2

In [165]:
sse <- t(e) %*% e
sse

In [166]:
sse_2 <- t(e_2) %*% e_2
sse_2

In [167]:
mse <- sse / 3
mse

In [168]:
mse_2 <- sse_2 / 3
mse_2

## (3)

$x_1 = x_2 = 2$에서 $E(y)$의 95% 신뢰구간을 구하시오.

In [169]:
new_dt <- data.frame(x1=2, x2=2)
new_dt

In [170]:
predict(m, newdata = new_dt,interval = c("confidence"), level = 0.95)

In [171]:
x0 <- c(1,2,2)

In [172]:
mu0 <- x0 %*% b

In [173]:
t(x0) %*% xx_i %*% x0

In [174]:
var_mu0 <- (t(x0) %*% xx_i %*% x0) * mse
var_mu0

In [175]:
tse <- qt(0.975,3) * sqrt(var_mu0)
tse

In [176]:
mu0 - (tse)

In [177]:
mu0 + (tse)

## (4)

$\beta_1$의 90% 신뢰구간을 구하시오.

In [178]:
summary(m)


Call:
lm(formula = y ~ ., data = dt)

Residuals:
      1       2       3       4       5       6 
-1.1395  1.3488  0.4651 -1.1628  1.4419 -0.9535 

Coefficients:
            Estimate Std. Error t value Pr(>|t|)  
(Intercept)   3.2326     1.9740   1.638   0.2000  
x1            1.5116     0.7713   1.960   0.1449  
x2           -2.6047     0.9288  -2.804   0.0676 .
---
Signif. codes:  0 ‘***’ 0.001 ‘**’ 0.01 ‘*’ 0.05 ‘.’ 0.1 ‘ ’ 1

Residual standard error: 1.599 on 3 degrees of freedom
Multiple R-squared:  0.7511,    Adjusted R-squared:  0.5852 
F-statistic: 4.527 on 2 and 3 DF,  p-value: 0.1242

In [179]:
confint(m, level = 0.9)

In [180]:
se_b1 <- sqrt(xx_i[2,2] * mse)
se_b1

In [181]:
tse <- qt(0.95,3) *se_b1
tse

In [182]:
b[2] - tse

In [183]:
b[2] + tse

## (5)

가설 $H_0: \beta_1 = \beta_2, H_1: \beta_1 \neq \beta_2$을 유의수준
$\alpha = 0.05$

In [184]:
library(car)

In [185]:
linearHypothesis(m, c(0,1,-1), 0)

In [186]:
z <- matrix(c(1,1,1,1,1,1,x[,2]+x[,3]), ncol=2)

In [187]:
zz <- t(z) %*% z
zz

In [188]:
zz_i <- solve(zz)
zz_i

In [189]:
zy <- t(z) %*% y
zy

In [190]:
yy <- t(y) %*% y
yy

In [191]:
z %*% zz_i %*% t(z)

In [192]:
gammahat <- zz_i %*% t(z) %*% y
gammahat

In [193]:
e <- y - z %*% gammahat
e

In [194]:
sse_rm <- t(e) %*% e
sse_rm

In [195]:
f0 = (sse_rm - sse)/(sse/3)
f0

In [196]:
qf(0.95,1,3)

10.1279644860139

**T검정 방법**

In [197]:
q = c(0,1,-1)

In [198]:
var_qb = t(q) %*% xx_i %*% q*  mse
var_qb

In [199]:
se_qb = sqrt(var_qb)
se_qb

In [200]:
t0 = (t(q) %*% b )/se_qb
t0

In [201]:
t0^2

## (6)

가설 $H_0 : \beta_1 = 0, H_1 : \beta_1 \neq 0$을 유의수준
$\alpha = 0.1$로 검정하시오. 위 (3)의 질문의 결과와 어떠한 연관을 지을
수 있는가?

In [202]:
t0 <- b[2] / se_b1

In [203]:
qt(0.95,3)

2.35336343480182

## (7)

$x_1 = 2, x_2 = 3$에서 가설 $H_0 : E(y) = 4,H_1 : E(y) \neq 4$를
$\alpha = 0.05$로 검정하시오

In [204]:
new_dt <- data.frame(x1=2, x2=2)

In [205]:
predict(m, newdata = new_dt,interval = c("confidence"), level = 0.95)

In [206]:
x0 <- c(1,2,3)

In [207]:
mu0 <- x0 %*% b

In [208]:
t(x0) %*% xx_i %*% x0

In [209]:
var_mu0 <- (t(x0) %*% xx_i %*% x0) * mse
var_mu0

In [210]:
t0 <- (mu0-4)/ sqrt(var_mu0)
t0

In [211]:
qt(0.975,3)

3.18244630528371

## (8)

가설 $H_0 : \beta_2 = 0, H_1: \beta_2 <0$을 $\alpha = 0.05$ 로
검정하시오.

In [212]:
se_b2 <- sqrt(xx_i[3,3] * mse)
se_b2

In [213]:
t0<- b[3] / se_b2
t0

## (9)

가설 \$H_0: \_1 = 2\_2, H_1 : \_1 \_2 \$를 $\alpha=0..05$로 검정하시오.

In [214]:
linearHypothesis(m, c(0,1,-2), 0)

In [215]:
z <- matrix(c(1,1,1,1,1,1,2*x[,2]+x[,3]), ncol=2)

In [216]:
zz <- t(z) %*% z
zz

In [217]:
zz_i <- solve(zz)
zz_i

In [218]:
zy <- t(z) %*% y
zy

In [219]:
yy <- t(y) %*% y
yy

In [220]:
z %*% zz_i %*% t(z)

In [221]:
gammahat <- zz_i %*% t(z) %*% y
gammahat

In [222]:
e <- y - z %*% gammahat
e

In [223]:
sse_rm <- t(e) %*% e
sse_rm

In [224]:
f0 = (sse_rm - sse)/(sse/3)
f0

In [225]:
qf(0.95,1,3)

10.1279644860139